### ISSUE TITLE
Guizmo position not centered to the origin of the object

### ISSUE THREAD

**Issue by LUK3D** [nariakiiwatani/ofxImGuizmo#2]
Hi!
Thanks for this awesome addon.
I'm using this addon with OF in a project that I working on for learning C++ (I know just the basics of c++).
The problem is when I added this to my project, it doesn't work. I have also tried the example in ofxImGui and in pfImGuizmo. Both of them have the same issue.

![image](https://github.com/nariakiiwatani/ofxImGuizmo/assets/31851972/b217548c-9432-4ae4-8fe4-dec1a20fda13)

![image](https://github.com/nariakiiwatani/ofxImGuizmo/assets/31851972/849bb167-f91d-4fe1-b545-c199384e7b4e)

How can I fix this if is something related to my setup?

**Comment by moebiussurfing** [comment]
Hey @LUK3D , @nariakiiwatani 

I think that the problem is here:
![image](https://github.com/nariakiiwatani/ofxImGuizmo/assets/2106149/216b5f2f-5a27-44f1-ac0c-f5a2dc2e6247)

Bc is getting the viewport rectangle from ImGui but there's no ImGui window, so I switched to get the viewport from OF.

![image](https://github.com/nariakiiwatani/ofxImGuizmo/assets/2106149/aba77413-7126-464f-9a1b-d227b67cd98f)


```.cpp

void ofApp::draw() {
	cam_.begin();
	{
		//node_.draw();

		ofPushMatrix();
		{
			node_.transformGL();  // Applies the node's transformation matrix

			ofPushStyle();
			ofNoFill();
			ofDrawBox(25);
			ofPopStyle();
		}
		ofPopMatrix();
	}
	cam_.end();

	//--

	gui_.begin();
	{
		ImGuizmo::BeginFrame();
		auto mat = node_.getLocalTransformMatrix();
		if (ImGuizmo::Manipulate(cam_, mat, op_, mode_)) {
			glm::mat4 transformation;
			glm::vec3 scale;
			glm::quat rotation;
			glm::vec3 translation;
			glm::vec3 skew;
			glm::vec4 perspective;
			glm::decompose(mat, scale, rotation, translation, skew, perspective);
			node_.setPosition(translation);
			node_.setScale(scale);
			node_.setOrientation(rotation);
		}
	}
	gui_.end();
}

```

**Comment by moebiussurfing** [comment]
You can check my modified fork, also with a new example that uses OF helpers:

https://github.com/moebiussurfing/ofxImGuizmo/blob/main/example-ImGuizmo/src/ofApp.cpp

Now the API is more simpler:

```.cpp
	cam_.begin();
	{
		//node_.draw();

		// Applies the node's transformation matrix
		ImGuizmo::beginGuizmoTransform(&node_);
		{
			ofPushStyle();
			ofNoFill();
			ofDrawBox(25);
			ofPopStyle();
		}
		ImGuizmo::endGuizmoTransform();
	}
	cam_.end();

	ImGuizmo::drawImGuizmo(&gui_, &cam_, &node_, op_, mode_);
```

**Comment by nariakiiwatani** [comment]
Thank you for everything!
so good, so nice!

for the viewport problem, I fixed it by getting viewport from ofCamera.  ( be0e0ce69bc552fa8a2d7ae0730bab423926f96d )
When I made this addon, I have used it inside ImWindow so the rectangle was implicitly same to oF viewport.

for simpler API, I think what @moebiussurfing wanted to do is essentially same to my second `Manipulate` functionality. 
https://github.com/nariakiiwatani/ofxImGuizmo/blob/main/src/ofxImGuizmo.h#L55-L62
I fixed the example using this. ( 9caba8b01b93da5ea34a640355e6ad61420ed299 )
